pkg/gateway/model_build_targets.go

```diff
@@ -171,14 +190,28 @@ func (t *latticeTargetsModelBuildTask) buildLatticeTargets(ctx context.Context)
 	return nil
 }
 
+func getDefinedPortsFromK8sRouteRules(t *latticeTargetsModelBuildTask, definedPorts []int64) []int64 {
+	if t.route != nil && t.route.Spec().Rules() != nil {
+		definedPorts = []int64{}
+		for _, rule := range t.route.Spec().Rules() {
+			for _, ref := range rule.BackendRefs() {
+				if ref.Port() != nil && string(ref.Name()) == t.tgName && string(*ref.Namespace()) == t.tgNamespace {
+					definedPorts = append(definedPorts, int64(*ref.Port()))
+				}
+			}
+		}
+	}
+	return definedPorts
+}
+
```

Comment:
good to see little functions with clear in-out, please add unit test for this

Reply:
This was removed now.